### PR TITLE
chore(api): add field selector api tests

### LINKED ISSUE #252: Add field selection tests to API Test
### Describe what should be investigated or refactored

Add additional API tests to test field selection

### PR DESCRIPTION
**Pull request by TristanHoladay**
## Description
Add api tests for field selectors

## Related Issue

Resolves #252

### CODE REVIEW

File: pkg/test/api_test.go [hunk 1]
```diff
@@ -195,6 +197,23 @@ func TestFieldSelectors(t *testing.T) {
 	}
 }
 
+func processResponseBody(body string) []byte {
```
Comment (UncleGedd) [inline pkg/test/api_test.go:200]: could you leave a comment mentioning how this fn makes the tests more stable?

Comment (decleaver) [reply pkg/test/api_test.go:200]: lol I read this as @UncleGedd being upset.... not as fn = function

File: pkg/test/api_test.go [hunk 2]
```diff
@@ -109,6 +111,109 @@ func TestQueryParams(t *testing.T) {
...
+
+func processResponseBody(body string) []byte {
+	// Regular expression to match "data: [...]"
+	re := regexp.MustCompile(`data: \[.*?\]`)
+
+	// Find all matches
+	matches := re.FindAllStringIndex(body, -1)
+
+	// Check if there are at least two matches
+	if len(matches) >= 2 {
```
Comment (decleaver) [inline pkg/test/api_test.go:208]: possible to make this more generic, and regardless of the number of matches, you can just grab the first one?

File: pkg/test/api_test.go [hunk 3]
```diff
@@ -109,10 +117,90 @@ func TestQueryParams(t *testing.T) {
 	}
 }
 
-type TestRoute struct {
-	name         string
-	url          string
-	expectedKind string
+// TODO: Add case for no ns with UID
```
Comment (decleaver) [inline pkg/test/api_test.go:120]: i feel like generally we dont want to leave "TODOs" in code, we can just do it or create issue for it?

**Review by decleaver — APPROVED** [review]
appreciate you addressing comments. Looks good

**Pull request merged**